### PR TITLE
Ignore tasks without data during project export

### PR DESCRIPTION
**Pull request by kirill-sizov**
### Motivation and context
- Allow backup and export annotations for a project even if it has tasks without data
- Add an error message on attempting to create backup of a task that has no data

### How has this been tested?
- Added REST API tests

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into the [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
~- [ ] I have increased versions of npm packages if it is necessary~
 ~([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

File: tests/python/rest_api/test_projects.py [hunk 1]
```diff
@@ -310,6 +310,27 @@ def test_org_owner_can_get_project_backup(
 
         self._test_can_get_project_backup(user["username"], project["id"])
 
+    def test_can_get_backup_project_when_some_tasks_have_no_data(self, projects):
+        project = next((p for p in projects if 0 < p["tasks"]["count"]))
+
+        # add empty task to project
+        post_method("admin1", "tasks", {"name": "empty_task", "project_id": project["id"]})
```
Comment (zhiltsov-max) [inline tests/python/rest_api/test_projects.py:317]: Probably, the response should be checked, as if the request fails, the test won't fail.

File: tests/python/rest_api/test_projects.py [hunk 2]
```diff
@@ -310,6 +310,27 @@ def test_org_owner_can_get_project_backup(
...
+
+        self._test_can_get_project_backup("admin1", project["id"])
+
+    def test_can_get_backup_project_when_all_tasks_have_no_data(self, projects):
+        project = next((p for p in projects if 0 == p["tasks"]["count"]))
+
+        # add empty tasks to empty project
+        post_method("admin1", "tasks", {"name": "empty_task1", "project_id": project["id"]})
+        post_method("admin1", "tasks", {"name": "empty_task2", "project_id": project["id"]})
```
Comment (zhiltsov-max) [inline tests/python/rest_api/test_projects.py:326]: As above

File: tests/python/rest_api/test_projects.py [hunk 3]
```diff
@@ -703,6 +724,27 @@ def test_can_export_dataset_with_skeleton_labels_with_spaces(self):
...
 
+    def test_can_export_dataset_for_empty_project(self, projects):
+        empty_project = next((p for p in projects if 0 == p["tasks"]["count"]))
+        self._test_export_project("admin1", empty_project["id"], "COCO 1.0")
+
+    def test_can_export_project_dataset_when_some_tasks_have_no_data(self, projects):
+        project = next((p for p in projects if 0 < p["tasks"]["count"]))
+
+        # add empty task to project
+        post_method("admin1", "tasks", {"name": "empty_task", "project_id": project["id"]})
```
Comment (zhiltsov-max) [inline tests/python/rest_api/test_projects.py:735]: As above

File: tests/python/rest_api/test_projects.py [hunk 4]
```diff
@@ -703,6 +724,27 @@ def test_can_export_dataset_with_skeleton_labels_with_spaces(self):
...
+
+        self._test_export_project("admin1", project["id"], "COCO 1.0")
+
+    def test_can_export_project_dataset_when_all_tasks_have_no_data(self, projects):
+        project = next((p for p in projects if 0 == p["tasks"]["count"]))
+
+        # add empty tasks to empty project
+        post_method("admin1", "tasks", {"name": "empty_task1", "project_id": project["id"]})
+        post_method("admin1", "tasks", {"name": "empty_task2", "project_id": project["id"]})
```
Comment (zhiltsov-max) [inline tests/python/rest_api/test_projects.py:744]: As above

File: cvat/apps/engine/views.py [hunk 5]
```diff
@@ -871,9 +871,15 @@ def append_backup_chunk(self, request, file_id):
             '200': OpenApiResponse(description='Download of file started'),
             '201': OpenApiResponse(description='Output backup file is ready for downloading'),
             '202': OpenApiResponse(description='Creating a backup file has been started'),
+            '400': OpenApiResponse(description='Backup of a task without data is not allowed'),
         })
     @action(methods=['GET'], detail=True, url_path='backup')
     def export_backup(self, request, pk=None):
+        if self.get_object().data is None:
+            return Response(
```
Comment (zhiltsov-max) [inline cvat/apps/engine/views.py:879]: Do we need a test?

Comment (kirill-sizov) [reply cvat/apps/engine/views.py:879]: Added

File: tests/python/rest_api/test_projects.py (hunk 1 above)
Comment (kirill-sizov) [reply tests/python/rest_api/test_projects.py]: Added response status check

**Review by zhiltsov-max — APPROVED** [review]

File: cvat/apps/dataset_manager/bindings.py [hunk 6]
```diff
@@ -949,7 +949,11 @@ def init(self):
 
     def _init_tasks(self):
         self._db_tasks: OrderedDict[int, Task] = OrderedDict(
-            ((db_task.id, db_task) for db_task in self._db_project.tasks.order_by("subset","id").all())
+            (
+                (db_task.id, db_task)
+                for db_task in self._db_project.tasks.order_by("subset","id").all()
```
Comment (nmanovic) [inline cvat/apps/dataset_manager/bindings.py:954]: @kirill-sizov , why not `.exclude(data=None)` as in other places?

**Pull request merged**